boards/nucleo-l073rz/include/periph_conf.h

```diff
@@ -95,23 +97,27 @@ static const uart_conf_t uart_config[] = {
         .rx_af      = GPIO_AF4,
         .tx_af      = GPIO_AF4,
         .bus        = APB2,
-        .irqn       = USART1_IRQn
+        .irqn       = USART1_IRQn,
+        .type       = STM32_USART,
+        .clk_src    = 0, /* Use APB clock */
     },
     {
-        .dev        = USART4,
-        .rcc_mask   = RCC_APB1ENR_USART4EN,
+        .dev        = LPUART1,
+        .rcc_mask   = RCC_APB1ENR_LPUART1EN,
         .rx_pin     = GPIO_PIN(PORT_C, 11),
         .tx_pin     = GPIO_PIN(PORT_C, 10),
-        .rx_af      = GPIO_AF6,
-        .tx_af      = GPIO_AF6,
+        .rx_af      = GPIO_AF0,
+        .tx_af      = GPIO_AF0,
         .bus        = APB1,
-        .irqn       = USART4_5_IRQn
+        .irqn       = LPUART1_IRQn,
+        .type       = STM32_LPUART,
+        .clk_src    = 0, /* Use APB clock */
```

Comment:
This change forces the driver to use lpuart, which is compiled and thus takes a bit more space than how it is in current master, e.g. with tests/periph_uart with this PR:

```console
  text	   data	    bss	    dec	    hex	filename
  15580	    504	   4752	  20836	   5164	/Users/facosta/git/RIOT-OS/RIOT/tests/periph_uart/bin/nucleo-l073rz/tests_periph_uart.elf
```

and with master:

```console
  text	   data	    bss	    dec	    hex	filename
  14892	    504	   4752	  20148	   4eb4	/Users/facosta/git/RIOT-OS/RIOT/tests/periph_uart/bin/nucleo-l073rz/tests_periph_uart.elf
```

With this PR on examples/hello-world:

```console
   text	   data	    bss	    dec	    hex	filename
   9252	    140	   2748	  12140	   2f6c	/Users/facosta/git/RIOT-OS/RIOT/examples/hello-world/bin/nucleo-l073rz/hello-world.elf
```

In current master:

```console
   text	   data	    bss	    dec	    hex	filename
   8564	    140	   2748	  11452	   2cbc	/Users/facosta/git/RIOT-OS/RIOT/examples/hello-world/bin/nucleo-l073rz/hello-world.elf
```

Thus, I'd vote to remove it but then it comes the problem on how to show/advertise the support of this feature? Can it be better a submodule of periph_uart?

Reply:
Your results make sense since the configuration of the nucleo-l073 board provides support for the 2 types of UARTs and thus more functions are built and included in the resulting firmware. But the ~700 bytes addition seems quite a lot and this seems weird, especially with the hello-world example. I'll look at it again.

Since this PR uses the same strategy implemented for the kinetis (the 2 types of UART drivers are implemented in the same file but one type can be skipped if not configured by the board), I would be interested to have the results of the build for one of them.

> Thus, I'd vote to remove it but then it comes the problem on how to show/advertise the support of this feature?

In fact, the nucleo-L073rz is the only L0 based board that is configured with an LPUART. This is not only useful for the given example but also for the CI, since it ensures the LPUART implementation is built at least for one L0 based board.

Reply:
> it ensures the LPUART implementation is built at least for one L0 based board.

Looking again at the changes, the lpuart functions were built for all L0/L4 based boards. I pushed an update to limit this to the boards that actually configure an LPUART, using a define.
Fixing the size increase with hello-world would be possible but would require a lot more work and is out of the scope of this PR: it would require the application to declare which UART is needed and then use some conditional macro to include or not the requested UART.

Reply:
Then the question remains: is it valid enough to increase our sizes by 700B for the sake of LPUART?

@bergzand is your tool to compare sizes between master and PRs still working?

Reply:
> is it valid enough to increase our sizes by 700B for the sake of LPUART?

From a CI point of view, I would say yes because now only this board provide an LPUART interface for L0. Having support for LPUART is also a nice feature for low-power applications.